
diff --git a/meson.build b/meson.build

--- a/meson.build	2022-03-13 02:03:28.000000000 +0300
+++ b/meson.build	2022-07-03 00:10:51.622845000 +0300
@@ -43,6 +43,15 @@
 gnome = import('gnome')
 i18n  = import('i18n')
 
+if get_option('local_install').auto()
+    gse_local_install = false
+    if get_option('prefix') == '/usr/local'
+        gse_local_install = run_command(find_program('id'), '-u').stdout().strip() != '0'
+    endif
+else
+    gse_local_install = get_option('local_install').enabled()
+endif
+
 gse_lib_convenience = files('meson-gse/lib/convenience.js')
 gse_lib_logger      = files('meson-gse/lib/logger.js')
 
@@ -69,6 +78,7 @@
   gse_schemas += files(gse_schema_main)
 endif
 
+gse_js91 = find_program('js91', required: false)
 gse_js78 = find_program('js78', required: false)
 gse_js68 = find_program('js68', required: false)
 gse_js60 = find_program('js60', required: false)
@@ -84,17 +94,28 @@
 # End of extension-specific settings
 
 # Boilerplate
-gse_run_command_obj = run_command('sh', '-c', 'echo $HOME')
-if gse_run_command_obj.returncode() != 0
-  error('HOME not found, exit=@0@'.format(gse_run_command_obj.returncode()))
-endif
-home     = gse_run_command_obj.stdout().strip()
-
-gse_uuid		 = meson.project_name() + '@extensions.gnome-shell.fifi.org'
-gse_target_dir		 = home + '/.local/share/gnome-shell/extensions/' + gse_uuid
-gse_target_dir_schemas   = join_paths(gse_target_dir, 'schemas')
-gse_target_locale_dir    = join_paths(gse_target_dir, 'locale')
-gse_target_dir_dbus_intf = join_paths(gse_target_dir, 'dbus-interfaces')
+if gse_local_install
+  gse_run_command_obj = run_command('sh', '-c', 'echo $HOME')
+  if gse_run_command_obj.returncode() != 0
+    error('HOME not found, exit=@0@'.format(gse_run_command_obj.returncode()))
+  endif
+  prefix = gse_run_command_obj.stdout().strip() / '.local'
+else
+  prefix = get_option('prefix')
+endif
+
+gse_uuid	= meson.project_name() + '@extensions.gnome-shell.fifi.org'
+gse_target_dir  = prefix + '/share/gnome-shell/extensions/' + gse_uuid
+
+if gse_local_install
+  gse_target_dir_schemas   = join_paths(gse_target_dir, 'schemas')
+  gse_target_locale_dir    = join_paths(gse_target_dir, 'locale')
+  gse_target_dir_dbus_intf = join_paths(gse_target_dir, 'dbus-interfaces')
+else
+  gse_target_dir_schemas = prefix / get_option('datadir') / 'glib-2.0' / 'schemas'
+  gse_target_locale_dir = prefix / get_option('localedir')
+  gse_target_dir_dbus_intf = join_paths(gse_target_dir, 'dbus-interfaces')
+endif
 
 meson_extra_scripts      = 'meson-gse/meson-scripts'
 
