diff -upr a/data/meson.build b/data/meson.build
--- a/data/meson.build	2023-03-19 15:50:13.898777755 +0300
+++ b/data/meson.build	2023-03-19 16:29:38.607135861 +0300
@@ -1,12 +1,14 @@
 data_conf = configuration_data()
 data_conf.set('libexecdir', libexecdir)
 
-configure_file(
-  input: 'power-profiles-daemon.service.in',
-  output: 'power-profiles-daemon.service',
-  configuration: data_conf,
-  install_dir: systemd_system_unit_dir,
-)
+if systemd_system_unit_dir != 'no'
+    configure_file(
+      input: 'power-profiles-daemon.service.in',
+      output: 'power-profiles-daemon.service',
+      configuration: data_conf,
+      install_dir: systemd_system_unit_dir,
+    )
+endif  
 
 configure_file(
   input: 'net.hadess.PowerProfiles.conf.in',
diff -upr a/meson.build b/meson.build
--- a/meson.build	2023-03-19 15:50:13.898777755 +0300
+++ b/meson.build	2023-03-19 16:16:50.064136551 +0300
@@ -27,9 +27,11 @@ dbusconfdir = get_option('datadir') / 'd
 dbusservicedir = get_option('datadir') / 'dbus-1' / 'system-services'
 
 systemd_system_unit_dir = get_option('systemdsystemunitdir')
-if systemd_system_unit_dir == 'auto'
-    systemd_dep = dependency('systemd')
-    systemd_system_unit_dir = systemd_dep.get_pkgconfig_variable('systemdsystemunitdir')
+if systemd_system_unit_dir != 'no'
+    if systemd_system_unit_dir == 'auto'
+        systemd_dep = dependency('systemd')
+        systemd_system_unit_dir = systemd_dep.get_pkgconfig_variable('systemdsystemunitdir')
+    endif
 endif
 gio_dep = dependency('gio-2.0')
 gudev_dep = dependency('gudev-1.0', version: '>= 234')
