
diff --git a/Makefile b/Makefile

--- a/Makefile	2022-03-12 14:06:41.000000000 +0300
+++ b/Makefile	2022-07-03 17:18:34.966377020 +0300
@@ -4,6 +4,13 @@
 
 PNG_FILES=$(wildcard ./docs/*.png)
 
+ifeq ($(strip $(DESTDIR)), )
+	INSTALL_DIR_TYPE = local
+else
+	INSTALL_DIR_TYPE = system
+	SHARE_PREFIX = $(DESTDIR)/usr/share
+endif
+
 .PHONY: build check prune compress install uninstall clean $(PNG_FILES)
 
 build:
@@ -20,8 +27,17 @@
 	$(MAKE) $(PNG_FILES)
 $(PNG_FILES):
 	optipng "$(COMPRESSLEVEL)" -strip all "$@"
-install:
+install: install-local
+
+install-local:
 	gnome-extensions install "$(UUID).shell-extension.zip" --force
+	
+ifeq ($(INSTALL_DIR_TYPE), system)
+	mkdir -p $(SHARE_PREFIX)/gnome-shell
+	mkdir -p $(SHARE_PREFIX)/gnome-shell/extensions
+	mv $(HOME)/.local/share/gnome-shell/extensions/$(UUID) $(SHARE_PREFIX)/gnome-shell/extensions/
+endif
+	
 uninstall:
 	gnome-extensions uninstall "$(UUID)"
 clean:
